MatrixSDKTests/MXSpaceServiceTest.swift

```diff
@@ -0,0 +1,159 @@
+//
+// Copyright 2021 The Matrix.org Foundation C.I.C
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+import XCTest
+
+import MatrixSDK
+
+class MXSpaceServiceTest: XCTestCase {
+    
+    // MARK: - Properties
+    
+    private var testData: MatrixSDKTestsData!
+    
+    // MARK: - Setup
+
+    override func setUpWithError() throws {
+        // Put setup code here. This method is called before the invocation of each test method in the class.
+        testData = MatrixSDKTestsData()
+    }
+
+    override func tearDownWithError() throws {
+        // Put teardown code here. This method is called after the invocation of each test method in the class.
+        testData = nil
+    }
+    
+    // MARK: - Private
+    
+    private func doSpaceServiceTestWithBob(testCase: XCTestCase, readyToTest: @escaping (_ spaceService: MXSpaceService, _ mxSession: MXSession, _ expectation: XCTestExpectation) -> Void) {
+        testData.doMXSessionTest(withBob: self) { (mxSession, expectation) in
+            guard let spaceService = mxSession?.spaceService else {
+                XCTFail("MXSession should have a spaceService instanciated")
+                return
+            }
+            readyToTest(spaceService, mxSession!, expectation!)
+        }
+    }
+    
+    private func waitRoomSummaryUpdate(for roomId: String, completion: @escaping ((MXRoomSummary) -> Void)) {
+        
+        var token: NSObjectProtocol?
+        
+        token = NotificationCenter.default.addObserver(forName: NSNotification.Name.mxRoomSummaryDidChange, object: nil, queue: OperationQueue.main, using: { notification in
```

Comment:
I wonder if we should put this logic in the SDK, either in MXRoom or MXSpace.

Reply:
In fact this method only wait for the first catch of room summary update notification. I don't know if it can be useful when using the SDK.